core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -0,0 +1,239 @@
+    private String strategy = PluginWrapper.PluginDisableStrategy.NONE.toString();
+
+    @Option(name = "-quiet", aliases = "-q", usage = "Be quiet, print only the error messages")
+    private boolean quiet;
+
+    private static final int INDENT_SPACE = 3;
+
+    @Override
+    public String getShortDescription() {
+        return Messages.DisablePluginCommand_ShortDescription();
+    }
+
+    // package-private access to be able to use it in the tests
+    static final int RETURN_CODE_NOT_DISABLED_DEPENDANTS = 16;
+    static final int RETURN_CODE_NO_SUCH_PLUGIN = 17;
+
+    @Override
+    protected void printUsageSummary(PrintStream stderr) {
+        super.printUsageSummary(stderr);
+        stderr.println(Messages.DisablePluginCommand_PrintUsageSummary());
+    }
+
+    @Override
+    protected int run() throws Exception {
+        Jenkins jenkins = Jenkins.get();
+        jenkins.checkPermission(Jenkins.ADMINISTER);
+
+        // get the strategy as an enum
+        PluginWrapper.PluginDisableStrategy strategyToUse;
+        try {
+            strategyToUse = PluginWrapper.PluginDisableStrategy.valueOf(strategy.toUpperCase());
+        } catch (IllegalArgumentException iae) {
+            throw new IllegalArgumentException(hudson.cli.Messages.DisablePluginCommand_NoSuchStrategy(strategy, String.format("%s, %s, %s", PluginWrapper.PluginDisableStrategy.NONE, PluginWrapper.PluginDisableStrategy.MANDATORY, PluginWrapper.PluginDisableStrategy.ALL)));
+        }
+
+        // disable...
+        List<PluginWrapper.PluginDisableResult> results = jenkins.pluginManager.disablePlugins(strategyToUse, pluginNames);
+
+        // print results ...
+        printResults(results);
+
+        // restart if it was required and it's necessary (at least one plugin was disabled in this execution)
+        restartIfNecessary(results);
+
+        // return the appropriate error code
+        return getResultCode(results);
+    }
+
+    /**
+     * Print the result of all the process
+     * @param results the list of results for the disablement of each plugin
+     */
+    private void printResults(List<PluginWrapper.PluginDisableResult> results) {
+        for (PluginWrapper.PluginDisableResult oneResult : results) {
+            printResult(oneResult, 0);
+        }
+    }
+
+    /**
+     * Print indented the arguments with the format passed beginning with the indent passed.
+     * @param indent number of spaces at the beginning.
+     * @param format format as in {@link String#format(String, Object...)}
+     * @param arguments arguments to print as in {@link String#format(String, Object...)}
+     */
+    private void printIndented(int indent, String format, String... arguments) {
+        if (indent == 0) {
+            stdout.format(format + "%n", (Object[]) arguments);
+        } else {
+            String f = "%" + indent + "s" + format + "%n";
+            String[] newArgs = new String[arguments.length + 1];
```

Comment:
I think there needs to be a more comprehensible way of doing this, it seems to complicated just to add a new string in the first position of an array.